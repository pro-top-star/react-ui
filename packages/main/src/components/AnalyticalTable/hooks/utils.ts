export const makeTemplateColumns = (columns, resizedColumns) => {
  return columns
    .map((column) => {
      if (resizedColumns[column.id]) {
        return `${resizedColumns[column.id]}px`;
      }
      return `minmax(${column.minWidth}px, ${column.width || '1fr'})`;
    })
    .join(' ');
};
