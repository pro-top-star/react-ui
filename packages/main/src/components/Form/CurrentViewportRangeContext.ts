import React from 'react';

const CurrentRange = React.createContext(null);

export { CurrentRange };
