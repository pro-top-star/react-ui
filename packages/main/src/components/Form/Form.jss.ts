import { JSSTheme } from '../../interfaces/JSSTheme';

const styles = ({ parameters }: JSSTheme) => ({
  formTitle: {
    borderBottom: `1px solid ${parameters.sapUiGroupTitleBorderColor}`
  },
  formTitlePaddingBottom: {
    paddingBottom: '2em'
  },
  formPaddingBottom: {
    paddingBottom: '1em'
  },
  formGroupStyle: {
    width: '100%',
    paddingTop: '0.25em'
  },
  formItemTopDiv: {
    alignItems: 'center'
  },
  formLabel: {
    paddingRight: '0.5em'
  },
  formElement: {
    display: 'block'
  }
});

export { styles };
