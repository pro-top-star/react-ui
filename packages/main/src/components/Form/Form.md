```jsx
import { Form } from '@ui5/webcomponents-react/lib/Form';
```
