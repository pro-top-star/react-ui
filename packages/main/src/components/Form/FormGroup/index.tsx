import { FlexBox } from '@ui5/webcomponents-react/lib/FlexBox';
import { FlexBoxAlignItems } from '@ui5/webcomponents-react/lib/FlexBoxAlignItems';
import { FlexBoxDirection } from '@ui5/webcomponents-react/lib/FlexBoxDirection';
import { FlexBoxJustifyContent } from '@ui5/webcomponents-react/lib/FlexBoxJustifyContent';
import { Title } from '@ui5/webcomponents-react/lib/Title';
import { TitleLevel } from '@ui5/webcomponents-react/lib/TitleLevel';
import React, { Children, FC, forwardRef, ReactNode, ReactNodeArray, Ref } from 'react';
import { createUseStyles } from 'react-jss';
import { styles } from '../Form.jss';

export interface FormGroupProps {
  title?: string;
  children: ReactNode | ReactNodeArray;
  type?: string;
}

const useStyles = createUseStyles<keyof ReturnType<typeof styles>>(styles, { name: 'FormGroup' });

const FormGroup: FC<FormGroupProps> = forwardRef((props: FormGroupProps, ref: Ref<HTMLDivElement>) => {
  const { title, children } = props;

  const classes = useStyles();

  return (
    <div ref={ref}>
      {title && (
        <Title level={TitleLevel.H5} className={classes.formPaddingBottom}>
          {title}
        </Title>
      )}
      <FlexBox
        justifyContent={FlexBoxJustifyContent.Start}
        alignItems={FlexBoxAlignItems.End}
        fitContainer
        direction={FlexBoxDirection.Column}
      >
        {Children.map(children, (child, index) => {
          return (
            <div key={index.toString()} className={classes.formGroupStyle}>
              {child}
            </div>
          );
        })}
      </FlexBox>
    </div>
  );
});

FormGroup.defaultProps = {
  type: 'formGroup'
};

export { FormGroup };
