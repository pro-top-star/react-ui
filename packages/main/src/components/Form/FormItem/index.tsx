import { CurrentViewportRangeContext } from '@ui5/webcomponents-react/lib/CurrentViewportRangeContext';
import { Label } from '@ui5/webcomponents-react/lib/Label';
import React, { FC, forwardRef, ReactNode, ReactNodeArray, Ref, useContext, useMemo } from 'react';
import { createUseStyles } from 'react-jss';
import { CommonProps } from '../../../interfaces/CommonProps';
import { JSSTheme } from '../../../interfaces/JSSTheme';
import { styles } from '../Form.jss';

export interface FormItemProps extends CommonProps {
  labelText?: string;
  children: ReactNode | ReactNodeArray;
}

const calculateWidth = (rate) => {
  return Math.floor((100 / 12) * rate) + '%';
};

const useStyles = createUseStyles<JSSTheme, keyof ReturnType<typeof styles>>(styles, { name: 'FormItem' });

/**
 * <code>import { FormItem } from '@ui5/webcomponents-react/lib/FormItem';</code>
 */
const FormItem: FC<FormItemProps> = forwardRef((props: FormItemProps, ref: Ref<HTMLDivElement>) => {
  const { labelText, children, tooltip, style, className, slot } = props;

  const currentRange = useContext(CurrentViewportRangeContext);

  const classes = useStyles();

  const memoizedStyles = useMemo(() => {
    let labelWidth;
    let labelTextAlign = 'flex-end';
    let display = 'flex';
    let elementWidth;

    switch (currentRange) {
      case 'Phone':
        labelWidth = '100%';
        elementWidth = '100%';
        display = 'block';
        labelTextAlign = 'flex-start';
        break;
      case 'Tablet':
        labelWidth = calculateWidth(2);
        elementWidth = calculateWidth(10);
        break;
      case 'Desktop':
      case 'LargeDesktop':
        labelWidth = calculateWidth(4);
        elementWidth = calculateWidth(8);
        break;
    }

    return {
      topDivStyle: {
        display: display,
        ...style
      },
      labelStyle: {
        width: labelWidth,
        justifyContent: labelTextAlign
      },
      elementStyle: {
        width: elementWidth
      }
    };
  }, [children, currentRange, style]);

  let classNames = `${classes.formItemTopDiv}`;
  if (className) {
    classNames += ` ${className}`;
  }

  return (
    <div ref={ref} style={memoizedStyles.topDivStyle} className={classNames} title={tooltip} slot={slot}>
      <Label style={memoizedStyles.labelStyle} className={classes.formLabel}>
        {labelText ? labelText : ''}
      </Label>
      <div style={memoizedStyles.elementStyle} className={classes.formElement}>
        {children}
      </div>
    </div>
  );
});

FormItem.displayName = 'FormItem';

export { FormItem };
