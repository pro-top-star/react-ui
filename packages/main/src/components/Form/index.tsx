import { usePassThroughHtmlProps } from '@ui5/webcomponents-react-base/lib/usePassThroughHtmlProps';
import { useViewportRange } from '@ui5/webcomponents-react-base/lib/useViewportRange';
import { Grid } from '@ui5/webcomponents-react/lib/Grid';
import { Title } from '@ui5/webcomponents-react/lib/Title';
import { TitleLevel } from '@ui5/webcomponents-react/lib/TitleLevel';
import React, { FC, forwardRef, ReactElement, ReactNode, ReactNodeArray, Ref, useMemo } from 'react';
import { createUseStyles } from 'react-jss';
import { CommonProps } from '../../interfaces/CommonProps';
import { JSSTheme } from '../../interfaces/JSSTheme';
import { CurrentRange } from './CurrentViewportRangeContext';
import { styles } from './Form.jss';
import { FormGroup } from './FormGroup';

export interface FormPropTypes extends CommonProps {
  /**
   * Components that are placed into Form.
   */
  children: ReactNode | ReactNodeArray;
  /**
   * Form title
   */
  title?: string;
}

const useStyles = createUseStyles<JSSTheme, keyof ReturnType<typeof styles>>(styles, { name: 'Form' });

const Form: FC<FormPropTypes> = forwardRef((props: FormPropTypes, ref: Ref<HTMLDivElement>) => {
  const { title, children } = props;

  const classes = useStyles();
  const currentRange = useViewportRange('StdExt');

  const [formGroups, updatedTitle] = useMemo(() => {
    let formGroups: any;
    let updatedTitle: string = title;

    // check if ungrouped FormItems exist amongst the Form's children and put them into an artificial FormGroup
    if (Array.isArray(children)) {
      const ungroupedItems = [];
      formGroups = [];
      children.forEach((child) => {
        if ((child as ReactElement).props.type === 'formItem') {
          ungroupedItems.push(child);
        } else if ((child as ReactElement).props.type === 'formGroup') {
          formGroups.push(child as ReactElement);
        }
      });

      if (ungroupedItems.length > 0) {
        formGroups.push(<FormGroup children={ungroupedItems} />);
      }
    } else {
      // check if a sole Form's group has a Title and take it as Form Title if one does not exist
      const childProps = (children as ReactElement).props;
      if ((!title || title.length === 0) && childProps.title && childProps.title.length > 0) {
        updatedTitle = childProps.title;
        formGroups = React.cloneElement(children as ReactElement, { title: null });
      } else {
        formGroups = children;
      }
    }

    return [formGroups, updatedTitle];
  }, [children]);

  const passThroughProps = usePassThroughHtmlProps(props);

  return (
    <CurrentRange.Provider value={currentRange}>
      {updatedTitle && (
        <>
          <Title level={TitleLevel.H3} className={classes.formTitle}>
            {updatedTitle}
          </Title>
          <div className={classes.formTitlePaddingBottom} />
        </>
      )}
      <Grid ref={ref} children={formGroups} defaultSpan={'XL6 L12 M12 S12'} {...passThroughProps} />
    </CurrentRange.Provider>
  );
});

Form.defaultProps = {
  children: [],
  title: null
};
Form.displayName = 'Form';

export { Form };
