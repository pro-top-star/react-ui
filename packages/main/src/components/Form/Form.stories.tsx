import { CheckBox } from '@ui5/webcomponents-react/lib/CheckBox';
import { Form } from '@ui5/webcomponents-react/lib/Form';
import { FormGroup } from '@ui5/webcomponents-react/lib/FormGroup';
import { FormItem } from '@ui5/webcomponents-react/lib/FormItem';
import { Input } from '@ui5/webcomponents-react/lib/Input';
import { InputType } from '@ui5/webcomponents-react/lib/InputType';
import { Option } from '@ui5/webcomponents-react/lib/Option';
import { Select } from '@ui5/webcomponents-react/lib/Select';
import React from 'react';

export const defaultStory = () => {
  return (
    <Form title={'Test Form'}>
      <FormGroup title={'Personal Data'}>
        <FormItem labelText={'Name'}>
          <Input type={InputType.Text} />
        </FormItem>
        <FormItem labelText={'Address'}>
          <Input type={InputType.Text} />
        </FormItem>
        <FormItem labelText={'Country'}>
          <Select>
            <Option>Germany</Option>
            <Option>France</Option>
            <Option>Italy</Option>
          </Select>
        </FormItem>
        <FormItem labelText={'Home address'}>
          <CheckBox checked />
        </FormItem>
      </FormGroup>
      <FormGroup title={'Business Data'}>
        <FormItem labelText={'Organization'}>
          <Input type={InputType.Text} />
        </FormItem>
        <FormItem labelText={'Position'}>
          <Input type={InputType.Text} />
        </FormItem>
        <FormItem labelText={'Wage'}>
          <Input type={InputType.Number} value={'5000'} disabled />
        </FormItem>
        <FormItem labelText={'Pilot license'}>
          <CheckBox checked />
        </FormItem>
      </FormGroup>
    </Form>
  );
};

export default {
  title: 'Components / Form',
  component: Form,
  parameters: {
    subcomponents: { FormGroup, FormItem }
  }
};
