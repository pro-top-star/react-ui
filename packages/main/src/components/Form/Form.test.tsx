import { createPassThroughPropsTest, mountThemedComponent } from '@shared/tests/utils';
import { Form } from '@ui5/webcomponents-react/lib/Form';
import { FormGroup } from '@ui5/webcomponents-react/lib/FormGroup';
import { FormItem } from '@ui5/webcomponents-react/lib/FormItem';
import * as React from 'react';
import { InputType } from '../..';
import { Input } from '../../webComponents/Input';

const SIZE_S = 200;
const SIZE_M = 800;
const SIZE_L = 1200;
const SIZE_XL = 1600;
const component = (
  <Form title={'Test form'}>
    <FormGroup title={'Group 1'}>
      <FormItem labelText={'item 1'}>
        <Input type={InputType.Text}></Input>
      </FormItem>
      <FormItem labelText={'item 2'}>
        <Input type={InputType.Number}></Input>
      </FormItem>
    </FormGroup>
    <FormGroup title={'Group 2'}>
      <FormItem labelText={'item 1'}>
        <Input type={InputType.Text}></Input>
      </FormItem>
      <FormItem labelText={'item 2'}>
        <Input type={InputType.Number}></Input>
      </FormItem>
    </FormGroup>
  </Form>
);

describe('Create a Form', () => {
  test('size rate S; should create Label and Element with 100% width and display: block for top FormItem div', () => {
    window = Object.assign(window, { innerWidth: SIZE_S });
    const wrapper = mountThemedComponent(component);
    expect(wrapper.render()).toMatchSnapshot();
  });

  test('size rate M; should create Label and Element with 16% and 83% width respectively and display: flex for top FormItem div', () => {
    window = Object.assign(window, { innerWidth: SIZE_M });
    const wrapper = mountThemedComponent(component);
    expect(wrapper.render()).toMatchSnapshot();
  });

  test('size rate L; should create Label and Element with 33% and 66% width respectively and display: flex for top FormItem div', () => {
    window = Object.assign(window, { innerWidth: SIZE_L });
    const wrapper = mountThemedComponent(component);
    expect(wrapper.render()).toMatchSnapshot();
  });

  test('size rate XL; should create Label and Element with 33% and 66% width respectively and display: flex for top FormItem div', () => {
    window = Object.assign(window, { innerWidth: SIZE_XL });
    const wrapper = mountThemedComponent(component);
    expect(wrapper.render()).toMatchSnapshot();
  });

  test('should create a FormGroup and put ungrouped FormItems into it', () => {
    const ungroupedChildren = (
      <Form title={'Test form'}>
        <FormItem labelText={'item 1'}>
          <Input type={InputType.Text}></Input>
        </FormItem>
        <FormItem labelText={'item 2'}>
          <Input type={InputType.Number}></Input>
        </FormItem>
      </Form>
    );
    const wrapper = mountThemedComponent(ungroupedChildren);
    expect(wrapper.render()).toMatchSnapshot();
  });

  test("should use a single FormGroup's title as a Form title if one is not set", () => {
    const ungroupedChildren = (
      <Form>
        <FormGroup title={'To be Form title'}>
          <FormItem labelText={'item 1'}>
            <Input type={InputType.Text}></Input>
          </FormItem>
          <FormItem labelText={'item 2'}>
            <Input type={InputType.Number}></Input>
          </FormItem>
        </FormGroup>
      </Form>
    );
    const wrapper = mountThemedComponent(ungroupedChildren);
    expect(wrapper.render()).toMatchSnapshot();
  });

  createPassThroughPropsTest(Form);
});
