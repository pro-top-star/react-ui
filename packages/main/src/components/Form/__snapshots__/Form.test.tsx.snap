// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create a Form should create a FormGroup and put ungrouped FormItems into it 1`] = `
Array [
  <ui5-title
    class="Form--formTitle-"
    level="H3"
  >
    Test form
  </ui5-title>,
  <div
    class="Form--formTitlePaddingBottom-"
  />,
  <div
    class="Grid--grid- Grid--gridHSpace1- Grid--gridVSpace1-"
  >
    <div
      class="Grid--gridSpan-"
    >
      <div>
        <div
          class="FlexBox--flexBox- FlexBox--flexBoxDirectionColumn- FlexBox--justifyContentStart- FlexBox--alignItemsEnd- FlexBox--flexWrapNoWrap-"
        >
          <div
            class="FormGroup--formGroupStyle-"
          >
            <div
              class="FormItem--formItemTopDiv-"
              style="display: flex;"
            >
              <ui5-label
                class="FormItem--formLabel-"
                for=""
                style="width: 33%; justify-content: flex-end;"
              >
                item 1
              </ui5-label>
              <div
                class="FormItem--formElement-"
                style="width: 66%;"
              >
                <ui5-input
                  type="Text"
                  value=""
                  value-state="None"
                />
              </div>
            </div>
          </div>
          <div
            class="FormGroup--formGroupStyle-"
          >
            <div
              class="FormItem--formItemTopDiv-"
              style="display: flex;"
            >
              <ui5-label
                class="FormItem--formLabel-"
                for=""
                style="width: 33%; justify-content: flex-end;"
              >
                item 2
              </ui5-label>
              <div
                class="FormItem--formElement-"
                style="width: 66%;"
              >
                <ui5-input
                  type="Number"
                  value=""
                  value-state="None"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`Create a Form should use a single FormGroup's title as a Form title if one is not set 1`] = `
Array [
  <ui5-title
    class="Form--formTitle-"
    level="H3"
  >
    To be Form title
  </ui5-title>,
  <div
    class="Form--formTitlePaddingBottom-"
  />,
  <div
    class="Grid--grid- Grid--gridHSpace1- Grid--gridVSpace1-"
  >
    <div
      class="Grid--gridSpan-"
    >
      <div>
        <div
          class="FlexBox--flexBox- FlexBox--flexBoxDirectionColumn- FlexBox--justifyContentStart- FlexBox--alignItemsEnd- FlexBox--flexWrapNoWrap-"
        >
          <div
            class="FormGroup--formGroupStyle-"
          >
            <div
              class="FormItem--formItemTopDiv-"
              style="display: flex;"
            >
              <ui5-label
                class="FormItem--formLabel-"
                for=""
                style="width: 33%; justify-content: flex-end;"
              >
                item 1
              </ui5-label>
              <div
                class="FormItem--formElement-"
                style="width: 66%;"
              >
                <ui5-input
                  type="Text"
                  value=""
                  value-state="None"
                />
              </div>
            </div>
          </div>
          <div
            class="FormGroup--formGroupStyle-"
          >
            <div
              class="FormItem--formItemTopDiv-"
              style="display: flex;"
            >
              <ui5-label
                class="FormItem--formLabel-"
                for=""
                style="width: 33%; justify-content: flex-end;"
              >
                item 2
              </ui5-label>
              <div
                class="FormItem--formElement-"
                style="width: 66%;"
              >
                <ui5-input
                  type="Number"
                  value=""
                  value-state="None"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`Create a Form size rate L; should create Label and Element with 33% and 66% width respectively and display: flex for top FormItem div 1`] = `
Array [
  <ui5-title
    class="Form--formTitle-"
    level="H3"
  >
    Test form
  </ui5-title>,
  <div
    class="Form--formTitlePaddingBottom-"
  />,
  <div
    class="Grid--grid- Grid--gridHSpace1- Grid--gridVSpace1-"
  >
    <div
      class="Grid--gridSpan-"
    >
      <div>
        <ui5-title
          class="FormGroup--formPaddingBottom-"
          level="H5"
        >
          Group 1
        </ui5-title>
        <div
          class="FlexBox--flexBox- FlexBox--flexBoxDirectionColumn- FlexBox--justifyContentStart- FlexBox--alignItemsEnd- FlexBox--flexWrapNoWrap-"
        >
          <div
            class="FormGroup--formGroupStyle-"
          >
            <div
              class="FormItem--formItemTopDiv-"
              style="display: flex;"
            >
              <ui5-label
                class="FormItem--formLabel-"
                for=""
                style="width: 33%; justify-content: flex-end;"
              >
                item 1
              </ui5-label>
              <div
                class="FormItem--formElement-"
                style="width: 66%;"
              >
                <ui5-input
                  type="Text"
                  value=""
                  value-state="None"
                />
              </div>
            </div>
          </div>
          <div
            class="FormGroup--formGroupStyle-"
          >
            <div
              class="FormItem--formItemTopDiv-"
              style="display: flex;"
            >
              <ui5-label
                class="FormItem--formLabel-"
                for=""
                style="width: 33%; justify-content: flex-end;"
              >
                item 2
              </ui5-label>
              <div
                class="FormItem--formElement-"
                style="width: 66%;"
              >
                <ui5-input
                  type="Number"
                  value=""
                  value-state="None"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="Grid--gridSpan-"
    >
      <div>
        <ui5-title
          class="FormGroup--formPaddingBottom-"
          level="H5"
        >
          Group 2
        </ui5-title>
        <div
          class="FlexBox--flexBox- FlexBox--flexBoxDirectionColumn- FlexBox--justifyContentStart- FlexBox--alignItemsEnd- FlexBox--flexWrapNoWrap-"
        >
          <div
            class="FormGroup--formGroupStyle-"
          >
            <div
              class="FormItem--formItemTopDiv-"
              style="display: flex;"
            >
              <ui5-label
                class="FormItem--formLabel-"
                for=""
                style="width: 33%; justify-content: flex-end;"
              >
                item 1
              </ui5-label>
              <div
                class="FormItem--formElement-"
                style="width: 66%;"
              >
                <ui5-input
                  type="Text"
                  value=""
                  value-state="None"
                />
              </div>
            </div>
          </div>
          <div
            class="FormGroup--formGroupStyle-"
          >
            <div
              class="FormItem--formItemTopDiv-"
              style="display: flex;"
            >
              <ui5-label
                class="FormItem--formLabel-"
                for=""
                style="width: 33%; justify-content: flex-end;"
              >
                item 2
              </ui5-label>
              <div
                class="FormItem--formElement-"
                style="width: 66%;"
              >
                <ui5-input
                  type="Number"
                  value=""
                  value-state="None"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`Create a Form size rate M; should create Label and Element with 16% and 83% width respectively and display: flex for top FormItem div 1`] = `
Array [
  <ui5-title
    class="Form--formTitle-"
    level="H3"
  >
    Test form
  </ui5-title>,
  <div
    class="Form--formTitlePaddingBottom-"
  />,
  <div
    class="Grid--grid- Grid--gridHSpace1- Grid--gridVSpace1-"
  >
    <div
      class="Grid--gridSpan-"
    >
      <div>
        <ui5-title
          class="FormGroup--formPaddingBottom-"
          level="H5"
        >
          Group 1
        </ui5-title>
        <div
          class="FlexBox--flexBox- FlexBox--flexBoxDirectionColumn- FlexBox--justifyContentStart- FlexBox--alignItemsEnd- FlexBox--flexWrapNoWrap-"
        >
          <div
            class="FormGroup--formGroupStyle-"
          >
            <div
              class="FormItem--formItemTopDiv-"
              style="display: flex;"
            >
              <ui5-label
                class="FormItem--formLabel-"
                for=""
                style="width: 16%; justify-content: flex-end;"
              >
                item 1
              </ui5-label>
              <div
                class="FormItem--formElement-"
                style="width: 83%;"
              >
                <ui5-input
                  type="Text"
                  value=""
                  value-state="None"
                />
              </div>
            </div>
          </div>
          <div
            class="FormGroup--formGroupStyle-"
          >
            <div
              class="FormItem--formItemTopDiv-"
              style="display: flex;"
            >
              <ui5-label
                class="FormItem--formLabel-"
                for=""
                style="width: 16%; justify-content: flex-end;"
              >
                item 2
              </ui5-label>
              <div
                class="FormItem--formElement-"
                style="width: 83%;"
              >
                <ui5-input
                  type="Number"
                  value=""
                  value-state="None"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="Grid--gridSpan-"
    >
      <div>
        <ui5-title
          class="FormGroup--formPaddingBottom-"
          level="H5"
        >
          Group 2
        </ui5-title>
        <div
          class="FlexBox--flexBox- FlexBox--flexBoxDirectionColumn- FlexBox--justifyContentStart- FlexBox--alignItemsEnd- FlexBox--flexWrapNoWrap-"
        >
          <div
            class="FormGroup--formGroupStyle-"
          >
            <div
              class="FormItem--formItemTopDiv-"
              style="display: flex;"
            >
              <ui5-label
                class="FormItem--formLabel-"
                for=""
                style="width: 16%; justify-content: flex-end;"
              >
                item 1
              </ui5-label>
              <div
                class="FormItem--formElement-"
                style="width: 83%;"
              >
                <ui5-input
                  type="Text"
                  value=""
                  value-state="None"
                />
              </div>
            </div>
          </div>
          <div
            class="FormGroup--formGroupStyle-"
          >
            <div
              class="FormItem--formItemTopDiv-"
              style="display: flex;"
            >
              <ui5-label
                class="FormItem--formLabel-"
                for=""
                style="width: 16%; justify-content: flex-end;"
              >
                item 2
              </ui5-label>
              <div
                class="FormItem--formElement-"
                style="width: 83%;"
              >
                <ui5-input
                  type="Number"
                  value=""
                  value-state="None"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`Create a Form size rate S; should create Label and Element with 100% width and display: block for top FormItem div 1`] = `
Array [
  <ui5-title
    class="Form--formTitle-"
    level="H3"
  >
    Test form
  </ui5-title>,
  <div
    class="Form--formTitlePaddingBottom-"
  />,
  <div
    class="Grid--grid- Grid--gridHSpace1- Grid--gridVSpace1-"
  >
    <div
      class="Grid--gridSpan-"
    >
      <div>
        <ui5-title
          class="FormGroup--formPaddingBottom-"
          level="H5"
        >
          Group 1
        </ui5-title>
        <div
          class="FlexBox--flexBox- FlexBox--flexBoxDirectionColumn- FlexBox--justifyContentStart- FlexBox--alignItemsEnd- FlexBox--flexWrapNoWrap-"
        >
          <div
            class="FormGroup--formGroupStyle-"
          >
            <div
              class="FormItem--formItemTopDiv-"
              style="display: block;"
            >
              <ui5-label
                class="FormItem--formLabel-"
                for=""
                style="width: 100%; justify-content: flex-start;"
              >
                item 1
              </ui5-label>
              <div
                class="FormItem--formElement-"
                style="width: 100%;"
              >
                <ui5-input
                  type="Text"
                  value=""
                  value-state="None"
                />
              </div>
            </div>
          </div>
          <div
            class="FormGroup--formGroupStyle-"
          >
            <div
              class="FormItem--formItemTopDiv-"
              style="display: block;"
            >
              <ui5-label
                class="FormItem--formLabel-"
                for=""
                style="width: 100%; justify-content: flex-start;"
              >
                item 2
              </ui5-label>
              <div
                class="FormItem--formElement-"
                style="width: 100%;"
              >
                <ui5-input
                  type="Number"
                  value=""
                  value-state="None"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="Grid--gridSpan-"
    >
      <div>
        <ui5-title
          class="FormGroup--formPaddingBottom-"
          level="H5"
        >
          Group 2
        </ui5-title>
        <div
          class="FlexBox--flexBox- FlexBox--flexBoxDirectionColumn- FlexBox--justifyContentStart- FlexBox--alignItemsEnd- FlexBox--flexWrapNoWrap-"
        >
          <div
            class="FormGroup--formGroupStyle-"
          >
            <div
              class="FormItem--formItemTopDiv-"
              style="display: block;"
            >
              <ui5-label
                class="FormItem--formLabel-"
                for=""
                style="width: 100%; justify-content: flex-start;"
              >
                item 1
              </ui5-label>
              <div
                class="FormItem--formElement-"
                style="width: 100%;"
              >
                <ui5-input
                  type="Text"
                  value=""
                  value-state="None"
                />
              </div>
            </div>
          </div>
          <div
            class="FormGroup--formGroupStyle-"
          >
            <div
              class="FormItem--formItemTopDiv-"
              style="display: block;"
            >
              <ui5-label
                class="FormItem--formLabel-"
                for=""
                style="width: 100%; justify-content: flex-start;"
              >
                item 2
              </ui5-label>
              <div
                class="FormItem--formElement-"
                style="width: 100%;"
              >
                <ui5-input
                  type="Number"
                  value=""
                  value-state="None"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`Create a Form size rate XL; should create Label and Element with 33% and 66% width respectively and display: flex for top FormItem div 1`] = `
Array [
  <ui5-title
    class="Form--formTitle-"
    level="H3"
  >
    Test form
  </ui5-title>,
  <div
    class="Form--formTitlePaddingBottom-"
  />,
  <div
    class="Grid--grid- Grid--gridHSpace1- Grid--gridVSpace1-"
  >
    <div
      class="Grid--gridSpan-"
    >
      <div>
        <ui5-title
          class="FormGroup--formPaddingBottom-"
          level="H5"
        >
          Group 1
        </ui5-title>
        <div
          class="FlexBox--flexBox- FlexBox--flexBoxDirectionColumn- FlexBox--justifyContentStart- FlexBox--alignItemsEnd- FlexBox--flexWrapNoWrap-"
        >
          <div
            class="FormGroup--formGroupStyle-"
          >
            <div
              class="FormItem--formItemTopDiv-"
              style="display: flex;"
            >
              <ui5-label
                class="FormItem--formLabel-"
                for=""
                style="width: 33%; justify-content: flex-end;"
              >
                item 1
              </ui5-label>
              <div
                class="FormItem--formElement-"
                style="width: 66%;"
              >
                <ui5-input
                  type="Text"
                  value=""
                  value-state="None"
                />
              </div>
            </div>
          </div>
          <div
            class="FormGroup--formGroupStyle-"
          >
            <div
              class="FormItem--formItemTopDiv-"
              style="display: flex;"
            >
              <ui5-label
                class="FormItem--formLabel-"
                for=""
                style="width: 33%; justify-content: flex-end;"
              >
                item 2
              </ui5-label>
              <div
                class="FormItem--formElement-"
                style="width: 66%;"
              >
                <ui5-input
                  type="Number"
                  value=""
                  value-state="None"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="Grid--gridSpan-"
    >
      <div>
        <ui5-title
          class="FormGroup--formPaddingBottom-"
          level="H5"
        >
          Group 2
        </ui5-title>
        <div
          class="FlexBox--flexBox- FlexBox--flexBoxDirectionColumn- FlexBox--justifyContentStart- FlexBox--alignItemsEnd- FlexBox--flexWrapNoWrap-"
        >
          <div
            class="FormGroup--formGroupStyle-"
          >
            <div
              class="FormItem--formItemTopDiv-"
              style="display: flex;"
            >
              <ui5-label
                class="FormItem--formLabel-"
                for=""
                style="width: 33%; justify-content: flex-end;"
              >
                item 1
              </ui5-label>
              <div
                class="FormItem--formElement-"
                style="width: 66%;"
              >
                <ui5-input
                  type="Text"
                  value=""
                  value-state="None"
                />
              </div>
            </div>
          </div>
          <div
            class="FormGroup--formGroupStyle-"
          >
            <div
              class="FormItem--formItemTopDiv-"
              style="display: flex;"
            >
              <ui5-label
                class="FormItem--formLabel-"
                for=""
                style="width: 33%; justify-content: flex-end;"
              >
                item 2
              </ui5-label>
              <div
                class="FormItem--formElement-"
                style="width: 66%;"
              >
                <ui5-input
                  type="Number"
                  value=""
                  value-state="None"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
